packages/router/history.ts

```diff
@@ -553,8 +567,12 @@ export function createURL(location: Location | string): URL {
     typeof window.location !== "undefined" &&
     window.location.origin !== "null"
       ? window.location.origin
-      : "unknown://unknown";
+      : window.location.href;
   let href = typeof location === "string" ? location : createPath(location);
+  invariant(
+    base,
+    `No window.location.(origin|href) available to create URL for href: ${href}`
+  );
   return new URL(href, base);
```

Comment:
The `unknown://unknown` was a bit of a snap decision very early on, but I think was really a red herring in some unit testing setup issues since `new URL('/', 'unknown://unknown')` doesn't actually work quite right anyway!  Now that we've better defined these APIs as intended only for client-side use - we can just use `invariant` if we can't detect a proper `base`.

For the specific Firefox `file://` case, `unknown://unknown` wasn't actually working right and gave incorrect paths - but it didn't throw an error.  Instead, firefox does have a "valid" `window.location.href` we can use as a fallback `base` when `origin = "null"` (in `file://` mode):

<img width="1247" alt="Screenshot 2022-12-05 at 4 21 11 PM" src="https://user-images.githubusercontent.com/1609022/205747207-dfb248d3-0cae-4ed8-a3d3-cf6926215954.png">